addons/website_livechat/tests/test_website_visitor.py

```diff
@@ -0,0 +1,44 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.addons.website.tests.test_website_visitor import WebsiteVisitorTests
+from odoo.tests import tagged
+
+
+@tagged('website_visitor')
+class WebsiteVisitorTestsLivechat(WebsiteVisitorTests):
+    def test_link_to_visitor_livechat(self):
+        """ Same as parent's 'test_link_to_visitor' except we also test that conversations
+        are merged into main visitor. """
+        [main_visitor, linked_visitor] = self.env['website.visitor'].create([
+            self._prepare_main_visitor_data(),
+            self._prepare_linked_visitor_data()
+        ])
+        linked_visitor._link_to_visitor(main_visitor)
+
+        self.assertVisitorDeactivated(linked_visitor, main_visitor)
+
+        # conversations of both visitors should be merged into main one
+        self.assertEqual(
+            main_visitor.mail_channel_ids.mapped('name'),
+            ['Conversation 1', 'Conversation 2'])
+
+    def _prepare_main_visitor_data(self):
```

Comment:
Github order makes [REF] website_event: move website.visitor's parent_id to website commit first, and with that commit _prepare_main_visitor_data does not exist (added in another commit). Not sure if wrong rebase of commits or simply bad ordering of commits.